api/elections.go

```diff
@@ -281,58 +281,84 @@ func (a *API) electionScrutinyHandler(msg *apirest.APIdata, ctx *httprouter.HTTP
 	if process.Status != models.ProcessStatus_RESULTS {
 		return fmt.Errorf("election results are not available yet")
 	}
-	// check process results are the same
-	var results [][]*big.Int
-	if len(process.Results) > 0 {
-		firstResult := process.Results[0]
-		for _, processResult := range process.Results[1:] {
-			if processResult == nil {
-				log.Errorf("invalid process result")
+	if process.Results == nil {
+		return fmt.Errorf("no results available")
+	}
+
+	// intermediateResults is an array of the results that are signed
+	// by oracles. There can be void results see vochain/state/process.go#L330
+	// but we want to ignore them. The way we can know if a result is void is
+	// by checking if the oracle address is empty.
+	intermediateResults := []*models.ProcessResult{}
+	for _, processResult := range process.Results {
+		if processResult == nil {
+			log.Warn("invalid process result (nil)")
+			continue
+		}
+		if len(processResult.OracleAddress) == 0 { // check oracle sent the results otherwise ignore
+			continue
+		}
+		intermediateResults = append(intermediateResults, processResult)
+	} // now we have an array of non void results set by a set of oracles via SetProcessResults
+
+	if len(intermediateResults) == 0 {
+		return fmt.Errorf("no results available")
+	}
+	firstResult := intermediateResults[0]
+
+	// Results are being compared because we want to make sure that
+	// all oracles are sending the same results.
+	equalResults := func(baseResult, otherResult *models.ProcessResult) bool {
+		// if the number of votes is different, the results are different
+		if len(baseResult.Votes) != len(otherResult.Votes) {
+			log.Warnf("different number of votes")
+			return false
+		}
+		for k, questionResult := range otherResult.Votes {
```

Comment:
to be consistent, you could name the parameters `result1, result2`, and then the votes `vote1, vote2` at the top of this loop. You're indexing `Votes[k]` in multiple places otherwise.